lib/dsprot/Makefile

```diff
@@ -0,0 +1,215 @@
+ifeq ($(NODEP),)
+  DEP_PARAM := -gccdep -MD
+  include $(wildcard $(C_OBJ_DIR)/*.d)
+else
+  DEP_PARAM := 
+endif
+
+INSTALL_PREFIX ?= ./install
+INSTALL_LIBDIR := $(INSTALL_PREFIX)/lib
+
+ELFCODER_DIR := $(TOOL_DIR)/elfcoder
+
+# Tools
+ELFCODER := $(ELFCODER_DIR)/build/elfcoder$(EXE)
+
+# C / ASM compilation parameters
+CC_PARAM   :=  -O4,p -enum int -proc arm946E -gccext,on -fp soft -lang c99 -char signed -inline on,noauto -Cpp_exceptions off -interworking -i $(INC_DIR) -c
+ASM_PARAM  :=  -proc arm5TE -i $(INC_DIR)
+LIB_PARAM  :=  -nostdlib -library
+
+CC_PARAM   +=  -W all -W pedantic -W noimpl_signedunsigned -W noimplicitconv -W nounusedarg -W nomissingreturn -W error
+
+# Output library file
+LIBRARY_NAME  :=  dsprot.a
+LIBRARY       :=  $(BUILD_DIR)/$(LIBRARY_NAME)
+
+# Files (in this specific order) that will go into the library
+LIBRARY_FILES := \
+	$(BUILD_DIR)/dsprot_main_encrypted.o          \
+	$(BUILD_DIR)/dsprot_main_decrypter_encoded.o  \
+	$(BUILD_DIR)/dsprot_main_decrypter_decoder.o  \
+	$(C_OBJ_DIR)/extra.o                          \
+	$(BUILD_DIR)/integrity_encrypted.o            \
+	$(BUILD_DIR)/integrity_decrypter_encoded.o    \
+	$(BUILD_DIR)/integrity_decrypter_decoder.o    \
+	$(BUILD_DIR)/encryptor_encoded.o              \
+	$(BUILD_DIR)/encryptor_decoder.o              \
+	$(BUILD_DIR)/mac_owner_encrypted.o            \
+	$(BUILD_DIR)/mac_owner_decrypter_encoded.o    \
+	$(BUILD_DIR)/rom_util_encrypted.o             \
+	$(BUILD_DIR)/rom_util_decrypter_encoded.o     \
+	$(BUILD_DIR)/rom_test_encrypted.o             \
+	$(BUILD_DIR)/rom_test_decrypter_encoded.o     \
+	$(BUILD_DIR)/coretests_decrypter_decoder.o    \
+	$(BUILD_DIR)/rc4_encoded.o                    \
+	$(BUILD_DIR)/rc4_decoder.o
+
+# Encryption keys
+KEY_DSPROT_MAIN := 6AB2
+KEY_INTEGRITY   := 9785
+KEY_CORE_TESTS  := 0982
+
+.PHONY: all clean tidy tools dsprot install
+.DELETE_ON_ERROR: 
+.NOTPARALLEL: 
+.SECONDARY: none
+
+all:
+	$(MAKE) tools
+	$(MAKE) dsprot
+
+clean:
+	$(MAKE) -C $(ELFCODER_DIR) clean
+	$(RM) -r $(BUILD_DIR)
+
+tidy: clean
+
+tools:
+	$(MAKE) -C $(ELFCODER_DIR)
+
+dsprot:
+	$(MAKE) $(LIBRARY)
+
+install:
+	$(MAKE) all
+	$(shell mkdir -p $(INSTALL_LIBDIR))
+	cp $(LIBRARY) $(INSTALL_LIBDIR)
+
+
+# Assembly assembling
+$(BUILD_DIR)/%.o: $(BUILD_DIR)/%.s
+	$(WINE) $(MWAS) $(ASM_PARAM) $< -o $@
+
+
+# C compilation
+$(C_OBJ_DIR)/%.o: $(SRC_DIR)/%.c
+	$(WINE) $(MWCC) $(CC_PARAM) $(DEP_PARAM) $< -o $@
+	@$(call fixdep,$(@:.o=.d))
+
+
+# Library output
+$(LIBRARY): $(LIBRARY_FILES)
+	$(WINE) $(MWLD) $(LIB_PARAM) $^ -o $@
+
+
+# Main module function encoding
+$(BUILD_DIR)/dsprot_main_decrypter_encoded.o \
+$(BUILD_DIR)/dsprot_main_decrypter_decoder.s: $(BUILD_DIR)/dsprot_main_decrypter.o $(ELFCODER)
+	cp $(BUILD_DIR)/dsprot_main_decrypter.o $(BUILD_DIR)/dsprot_main_decrypter_encoded.o
+	$(ELFCODER) -e -i $(BUILD_DIR)/dsprot_main_decrypter_encoded.o -o $(BUILD_DIR)/dsprot_main_decrypter_decoder.s -g Garbage -f \
+		DSProt_DetectFlashcart     \
```

Comment:
is this necessary to split over lines like this
is there not a better way to do this?
I'm not sure how functions are chosen to be encrypted but I'm a bit... cautious about having stuff like this hardcoded
makes things difficult to change/update later
(yes I know it's a protection library but you never know)